models/spike_train_injector.h

```diff
@@ -0,0 +1,505 @@
+/*
+ *  spike_train_injector.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef SPIKE_TRAIN_INJECTOR_H
+#define SPIKE_TRAIN_INJECTOR_H
+
+// C++ includes:
+#include <vector>
+
+// Includes from nestkernel:
+#include "connection.h"
+#include "event.h"
+#include "nest_time.h"
+#include "nest_types.h"
+#include "node.h"
+
+namespace nest
+{
+
+/* BeginUserDocs: neuron, spike train injector
+
+Short description
++++++++++++++++++
+
+Neuron that emits prescribed spike trains.
+
+Description
++++++++++++
+
+The spike train injector neuron simply emits spikes at prescribed spike times
+which are given as an array. Incoming spikes will be ignored.
+
+The spike train injector neuron behaves similarly to a spike generator, but
```

Comment:
This and the next section should be much shorter and less technical, just a recommendation: use this one if you spike trains have similar rate to normal neurons, use spike_generator if you have very high rates.